django/cantusdb_project/main_app/urls.py

```diff
@@ -58,10 +59,11 @@
         name="source-create"
     ),
     path(
-        "edit-volpiano/", 
+        "edit-volpiano/<int:source_id>", 
         ChantEditVolpianoView.as_view(), 
         name="source-edit-volpiano"
     ),
+    path("users/<int:pk>", UserDetailView.as_view(), name="user-detail"),
```

Comment:
It would be better to give this `pk` a more informative name like `user_id`, just like what you did with the `source_id` above.